onmt/trainer.py

```diff
@@ -49,7 +49,7 @@ def build_trainer(opt, device_id, model, fields, optim, model_saver=None):
     if device_id >= 0:
         gpu_rank = opt.gpu_ranks[device_id]
     else:
-        gpu_rank = 0
+        gpu_rank = -1
```

Comment:
Hmm, not sure it's a good idea to use this value like that, but it probably doesn't hurt.

Reply:
Cause for the case of `device_id` < 0, it's cpu rather gpu, but if we use 0 that will be the NO.0 gpu. Semantically not identical. Also, I previously use this to do the batch transfer. But later find a better way. Anyway, this do not hurt anything.